### PR TITLE
UCT/IB: Relaxed order support

### PR DESCRIPTION
**Pull request by Artemy-Mellanox**

### CODE REVIEW

File: src/uct/ib/base/ib_md.c [hunk 1]
```diff
@@ -1603,6 +1612,12 @@ static ucs_status_t uct_ib_verbs_md_open(struct ibv_device *ibv_device,
         goto err_free_context;
     }
 
+    if (md->config.mr_relaxed_order == UCS_YES && !IBV_ACCESS_RELAXED_ORDERING) {
```
Comment (yosefe) [inline src/uct/ib/base/ib_md.c]: ( )

File: src/uct/ib/base/ib_md.c [hunk 2]
```diff
@@ -185,6 +185,10 @@ static ucs_config_field_t uct_ib_md_config_table[] = {
      "Enable setting CPU affinity of memory registration threads.",
      ucs_offsetof(uct_ib_md_config_t, ext.mt_reg_bind), UCS_CONFIG_TYPE_BOOL},
 
+    {"MEM_RELAXED_ORDERING", "try",
+     "Allow system to reorder memory accesses to improve performance.\n",
+     ucs_offsetof(uct_ib_md_config_t, ext.mr_relaxed_order), UCS_CONFIG_TYPE_TERNARY},
```
Comment (yosefe) [inline src/uct/ib/base/ib_md.c]: can you pls use UCS_CONFIG_TYPE_ON_OFF_AUTO
PCI_RELAXED_ORDER
"Enable relaxed ordering for PCI writes from device to memory, to improve performance on some systems."

File: src/uct/ib/base/ib_md.h [hunk 3]
```diff
@@ -83,6 +83,8 @@ typedef struct uct_ib_md_ext_config {
     size_t                   min_mt_reg;   /**< Multi-threaded registration threshold */
     size_t                   mt_reg_chunk; /**< Multi-threaded registration chunk */
     int                      mt_reg_bind;  /**< Multi-threaded registration bind to core */
+
+    int                      mr_relaxed_order; /**< Allow reorder memory accesses */
```
Comment (yosefe) [inline src/uct/ib/base/ib_md.h]: ucs_on_off_auto_value_t

**Comment by mellanox-github** [comment]
<details><summary>Mellanox CI: <b>PASSED</b#  > on 25 workers (click for details)</summary>

**Note:** the logs will be deleted after 08-Mar-2020

Agent/Stage | Status
-- | --
[_main](https://gist.githubusercontent.com/mellanox-github/46e85204236b1829bd15ae7ff73f6d9d/raw/293b35cc7c1d7f9c48cfc586da5ffd191112f626/_main) | :heavy_check_mark: SUCCESS
[hpc-arm-cavium-jenkins_W0](https://gist.githubusercontent.com/mellanox-github/46e85204236b1829bd15ae7ff73f6d9d/raw/9d8755daae94db7d211f1c294f746fb324b2778e/hpc-arm-cavium-jenkins_W0) | :heavy_check_mark: SUCCESS
[hpc-arm-cavium-jenkins_W1](https://gist.githubusercontent.com/mellanox-github/46e85204236b1829bd15ae7ff73f6d9d/raw/864a3fb035c8103975bde9b165448988eb0f56a8/hpc-arm-cavium-jenkins_W1) | :heavy_check_mark: SUCCESS
[hpc-arm-cavium-jenkins_W2](https://gist.githubusercontent.com/mellanox-github/46e85204236b1829bd15ae7ff73f6d9d/raw/248e5182f1fbeecf9e91821c9d5faaa2c11c3b6b/hpc-arm-cavium-jenkins_W2) | :heavy_check_mark: SUCCESS
[hpc-arm-cavium-jenkins_W3](https://gist.githubusercontent.com/mellanox-github/46e85204236b1829bd15ae7ff73f6d9d/raw/50c26f47c911df692dc08324808c7a845b1d8f0b/hpc-arm-cavium-jenkins_W3) | :heavy_check_mark: SUCCESS
[hpc-arm-hwi-jenkins_W0](https://gist.githubusercontent.com/mellanox-github/46e85204236b1829bd15ae7ff73f6d9d/raw/315030d117040860604b5c496fbe055c37ef6b78/hpc-arm-hwi-jenkins_W0) | :heavy_check_mark: SUCCESS
[hpc-arm-hwi-jenkins_W1](https://gist.githubusercontent.com/mellanox-github/46e85204236b1829bd15ae7ff73f6d9d/raw/a9216dee8365977163e724559741174334eb3059/hpc-arm-hwi-jenkins_W1) | :heavy_check_mark: SUCCESS
[hpc-arm-hwi-jenkins_W2](https://gist.githubusercontent.com/mellanox-github/46e85204236b1829bd15ae7ff73f6d9d/raw/a2a69849f647d8f965362e06a107afbf95684bbb/hpc-arm-hwi-jenkins_W2) | :heavy_check_mark: SUCCESS
[hpc-arm-hwi-jenkins_W3](https://gist.githubusercontent.com/mellanox-github/46e85204236b1829bd15ae7ff73f6d9d/raw/9a716b940029e23bdc9e7e0ef058abfd1ffc7994/hpc-arm-hwi-jenkins_W3) | :heavy_check_mark: SUCCESS
[hpc-test-node-gpu_W0](https://gist.githubusercontent.com/mellanox-github/46e85204236b1829bd15ae7ff73f6d9d/raw/2dadfd58c08163b2e27d23a346649ee47bbe9a51/hpc-test-node-gpu_W0) | :heavy_check_mark: SUCCESS
[hpc-test-node-gpu_W1](https://gist.githubusercontent.com/mellanox-github/46e85204236b1829bd15ae7ff73f6d9d/raw/8cac1937853fd988c60e7508ac09304ca54fa434/hpc-test-node-gpu_W1) | :heavy_check_mark: SUCCESS
[hpc-test-node-gpu_W2](https://gist.githubusercontent.com/mellanox-github/46e85204236b1829bd15ae7ff73f6d9d/raw/f5905f32e5da5ac3b9698377e1a312bee367bbce/hpc-test-node-gpu_W2) | :heavy_check_mark: SUCCESS
[hpc-test-node-gpu_W3](https://gist.githubusercontent.com/mellanox-github/46e85204236b1829bd15ae7ff73f6d9d/raw/23e18a0c89c8779826459da39f8f55dfaf370837/hpc-test-node-gpu_W3) | :heavy_check_mark: SUCCESS
[hpc-test-node-legacy_W0](https://gist.githubusercontent.com/mellanox-github/46e85204236b1829bd15ae7ff73f6d9d/raw/3f0b68cebddc6fe62522b987e772e0662df099c3/hpc-test-node-legacy_W0) | :heavy_check_mark: SUCCESS
[hpc-test-node-legacy_W1](https://gist.githubusercontent.com/mellanox-github/46e85204236b1829bd15ae7ff73f6d9d/raw/ca4108fd1447994b6e8abcf33afa826d2e0f3bd1/hpc-test-node-legacy_W1) | :heavy_check_mark: SUCCESS
[hpc-test-node-legacy_W2](https://gist.githubusercontent.com/mellanox-github/46e85204236b1829bd15ae7ff73f6d9d/raw/4738d7a00f847e6c312df5f4f0f392835345ec5e/hpc-test-node-legacy_W2) | :heavy_check_mark: SUCCESS
[hpc-test-node-legacy_W3](https://gist.githubusercontent.com/mellanox-github/46e85204236b1829bd15ae7ff73f6d9d/raw/bfa8ed97d3047f8e472cae2f47a44eefb5013429/hpc-test-node-legacy_W3) | :heavy_check_mark: SUCCESS
[hpc-test-node-new_W0](https://gist.githubusercontent.com/mellanox-github/46e85204236b1829bd15ae7ff73f6d9d/raw/a0a1f3a71d7f0075b22b70bc8937b4661679a999/hpc-test-node-new_W0) | :heavy_check_mark: SUCCESS
[hpc-test-node-new_W1](https://gist.githubusercontent.com/mellanox-github/46e85204236b1829bd15ae7ff73f6d9d/raw/4165d6e921c317a4c3a788f8f5ecca8c5687723d/hpc-test-node-new_W1) | :heavy_check_mark: SUCCESS
[hpc-test-node-new_W2](https://gist.githubusercontent.com/mellanox-github/46e85204236b1829bd15ae7ff73f6d9d/raw/af1a3808dd1ca8e6387fd403168fc805837f6f1f/hpc-test-node-new_W2) | :heavy_check_mark: SUCCESS
[hpc-test-node-new_W3](https://gist.githubusercontent.com/mellanox-github/46e85204236b1829bd15ae7ff73f6d9d/raw/c7f73c06da98f7239901ec140550433391d08c15/hpc-test-node-new_W3) | :heavy_check_mark: SUCCESS
[r-vmb-ppc-jenkins_W0](https://gist.githubusercontent.com/mellanox-github/46e85204236b1829bd15ae7ff73f6d9d/raw/1351bbf6b48e142024207996e4ac90060df1dd44/r-vmb-ppc-jenkins_W0) | :heavy_check_mark: SUCCESS
[r-vmb-ppc-jenkins_W1](https://gist.githubusercontent.com/mellanox-github/46e85204236b1829bd15ae7ff73f6d9d/raw/0de9fd0ef97c5ab71d9ce3c80005ef2bba9c3fb5/r-vmb-ppc-jenkins_W1) | :heavy_check_mark: SUCCESS
[r-vmb-ppc-jenkins_W2](https://gist.githubusercontent.com/mellanox-github/46e85204236b1829bd15ae7ff73f6d9d/raw/080cd7bf16639773d111afbb2f4742bc0482a44b/r-vmb-ppc-jenkins_W2) | :heavy_check_mark: SUCCESS
[r-vmb-ppc-jenkins_W3](https://gist.githubusercontent.com/mellanox-github/46e85204236b1829bd15ae7ff73f6d9d/raw/231528591809379d28f62ff1ff639ae57ca178ae/r-vmb-ppc-jenkins_W3) | :heavy_check_mark: SUCCESS

**Comment by mellanox-github** [comment]
<details><summary>Mellanox CI: <b>PASSED</b#  > on 25 workers (click for details)</summary>

**Note:** the logs will be deleted after 08-Mar-2020

Agent/Stage | Status
-- | --
[_main](https://gist.githubusercontent.com/mellanox-github/267d0649e87eab380fb3ebc1db192fc7/raw/900a0fb5e9bcb50f39ef08c5ee97a4229d901f57/_main) | :heavy_check_mark: SUCCESS
[hpc-arm-cavium-jenkins_W0](https://gist.githubusercontent.com/mellanox-github/267d0649e87eab380fb3ebc1db192fc7/raw/6ee50bc37e0f43ecd7277a28c9e770cf22a54d2e/hpc-arm-cavium-jenkins_W0) | :heavy_check_mark: SUCCESS
[hpc-arm-cavium-jenkins_W1](https://gist.githubusercontent.com/mellanox-github/267d0649e87eab380fb3ebc1db192fc7/raw/95ed3951c80d5a811cec56e8cff52585039dbb8f/hpc-arm-cavium-jenkins_W1) | :heavy_check_mark: SUCCESS
[hpc-arm-cavium-jenkins_W2](https://gist.githubusercontent.com/mellanox-github/267d0649e87eab380fb3ebc1db192fc7/raw/83dee550d985179e6e896e4ead524e7feef5e7ce/hpc-arm-cavium-jenkins_W2) | :heavy_check_mark: SUCCESS
[hpc-arm-cavium-jenkins_W3](https://gist.githubusercontent.com/mellanox-github/267d0649e87eab380fb3ebc1db192fc7/raw/8f39d25792c91850a07c9793cfe1e1df385aea7e/hpc-arm-cavium-jenkins_W3) | :heavy_check_mark: SUCCESS
[hpc-arm-hwi-jenkins_W0](https://gist.githubusercontent.com/mellanox-github/267d0649e87eab380fb3ebc1db192fc7/raw/548182358ca965cad67ee9adb125f2f2f868f1a7/hpc-arm-hwi-jenkins_W0) | :heavy_check_mark: SUCCESS
[hpc-arm-hwi-jenkins_W1](https://gist.githubusercontent.com/mellanox-github/267d0649e87eab380fb3ebc1db192fc7/raw/fcbfc5c8d61f62fe3edd443652119967894ee5dd/hpc-arm-hwi-jenkins_W1) | :heavy_check_mark: SUCCESS
[hpc-arm-hwi-jenkins_W2](https://gist.githubusercontent.com/mellanox-github/267d0649e87eab380fb3ebc1db192fc7/raw/6e6e8eab70e3ac8cf117ac77d0d1e78bf83ec4a9/hpc-arm-hwi-jenkins_W2) | :heavy_check_mark: SUCCESS
[hpc-arm-hwi-jenkins_W3](https://gist.githubusercontent.com/mellanox-github/267d0649e87eab380fb3ebc1db192fc7/raw/9d3ac6e85ee6e0eb5625c1c881369ed888c95ad7/hpc-arm-hwi-jenkins_W3) | :heavy_check_mark: SUCCESS
[hpc-test-node-gpu_W0](https://gist.githubusercontent.com/mellanox-github/267d0649e87eab380fb3ebc1db192fc7/raw/559af17d0658a1527cb70ed797dc820fd346a4e1/hpc-test-node-gpu_W0) | :heavy_check_mark: SUCCESS
[hpc-test-node-gpu_W1](https://gist.githubusercontent.com/mellanox-github/267d0649e87eab380fb3ebc1db192fc7/raw/b9bd8b8c6a59aca8367c8193496959d97e34933b/hpc-test-node-gpu_W1) | :heavy_check_mark: SUCCESS
[hpc-test-node-gpu_W2](https://gist.githubusercontent.com/mellanox-github/267d0649e87eab380fb3ebc1db192fc7/raw/fddd42e908f52c2685927cb43e5d2d7a90451346/hpc-test-node-gpu_W2) | :heavy_check_mark: SUCCESS
[hpc-test-node-gpu_W3](https://gist.githubusercontent.com/mellanox-github/267d0649e87eab380fb3ebc1db192fc7/raw/9c2113b1c95c9951b6929ace77fbbcc6b31ae91e/hpc-test-node-gpu_W3) | :heavy_check_mark: SUCCESS
[hpc-test-node-legacy_W0](https://gist.githubusercontent.com/mellanox-github/267d0649e87eab380fb3ebc1db192fc7/raw/6319d0a82bf20ecfeda43c44c325def905346335/hpc-test-node-legacy_W0) | :heavy_check_mark: SUCCESS
[hpc-test-node-legacy_W1](https://gist.githubusercontent.com/mellanox-github/267d0649e87eab380fb3ebc1db192fc7/raw/f8a611757f6da959522e6711a745c5f0a2b2cfd0/hpc-test-node-legacy_W1) | :heavy_check_mark: SUCCESS
[hpc-test-node-legacy_W2](https://gist.githubusercontent.com/mellanox-github/267d0649e87eab380fb3ebc1db192fc7/raw/dd8cade297f05c262d6a8f55268c457b40b1200d/hpc-test-node-legacy_W2) | :heavy_check_mark: SUCCESS
[hpc-test-node-legacy_W3](https://gist.githubusercontent.com/mellanox-github/267d0649e87eab380fb3ebc1db192fc7/raw/d2c1fd22c1f5d79f41ac5ad7a30236b4b9e5fa5c/hpc-test-node-legacy_W3) | :heavy_check_mark: SUCCESS
[hpc-test-node-new_W0](https://gist.githubusercontent.com/mellanox-github/267d0649e87eab380fb3ebc1db192fc7/raw/c9551a47b4c7cd36ec128f03ad16c54033807962/hpc-test-node-new_W0) | :heavy_check_mark: SUCCESS
[hpc-test-node-new_W1](https://gist.githubusercontent.com/mellanox-github/267d0649e87eab380fb3ebc1db192fc7/raw/8034098536e77bd066f5edb3ebc0f65451eb1be5/hpc-test-node-new_W1) | :heavy_check_mark: SUCCESS
[hpc-test-node-new_W2](https://gist.githubusercontent.com/mellanox-github/267d0649e87eab380fb3ebc1db192fc7/raw/7142928856af32e4e485a13307e5963c97997ab1/hpc-test-node-new_W2) | :heavy_check_mark: SUCCESS
[hpc-test-node-new_W3](https://gist.githubusercontent.com/mellanox-github/267d0649e87eab380fb3ebc1db192fc7/raw/d3d8341cab9a95ab75e8e0f66b482286bd7582ed/hpc-test-node-new_W3) | :heavy_check_mark: SUCCESS
[r-vmb-ppc-jenkins_W0](https://gist.githubusercontent.com/mellanox-github/267d0649e87eab380fb3ebc1db192fc7/raw/e41a4c7f7ec96cb400337856a0eab7ad2c66f33a/r-vmb-ppc-jenkins_W0) | :heavy_check_mark: SUCCESS
[r-vmb-ppc-jenkins_W1](https://gist.githubusercontent.com/mellanox-github/267d0649e87eab380fb3ebc1db192fc7/raw/7c0b7190037e72de6a69c80fca235e98dab8da32/r-vmb-ppc-jenkins_W1) | :heavy_check_mark: SUCCESS
[r-vmb-ppc-jenkins_W2](https://gist.githubusercontent.com/mellanox-github/267d0649e87eab380fb3ebc1db192fc7/raw/6db7b1b653d1ac7136cfab4fcf4d38e00c1359cc/r-vmb-ppc-jenkins_W2) | :heavy_check_mark: SUCCESS
[r-vmb-ppc-jenkins_W3](https://gist.githubusercontent.com/mellanox-github/267d0649e87eab380fb3ebc1db192fc7/raw/12d9b0e559b818fa19350cb1661f6f007b6d5c62/r-vmb-ppc-jenkins_W3) | :heavy_check_mark: SUCCESS

File: src/uct/ib/base/ib_md.c [hunk 4]
```diff
@@ -1603,6 +1613,12 @@ static ucs_status_t uct_ib_verbs_md_open(struct ibv_device *ibv_device,
         goto err_free_context;
     }
 
+    if ((md->config.mr_relaxed_order == UCS_CONFIG_ON) && !IBV_ACCESS_RELAXED_ORDERING) {
+        ucs_warn("relaxed oreder memory access requested but not supported");
```
Comment (brminich) [inline src/uct/ib/base/ib_md.c]: `order`

File: src/uct/ib/base/ib_md.c [hunk 5]
```diff
@@ -185,6 +185,11 @@ static ucs_config_field_t uct_ib_md_config_table[] = {
      "Enable setting CPU affinity of memory registration threads.",
      ucs_offsetof(uct_ib_md_config_t, ext.mt_reg_bind), UCS_CONFIG_TYPE_BOOL},
 
+    {"PCI_RELAXED_ORDERING", "auto",
```
Comment (brminich) [inline src/uct/ib/base/ib_md.c]: looks like it is off by default. Maybe set `off` as a default value?

Comment (yosefe) [reply src/uct/ib/base/ib_md.c]: we want to make "auto" detect by platform type

**Comment by mellanox-github** [comment]
<details><summary>Mellanox CI: <b>PASSED</b#  > on 25 workers (click for details)</summary>

**Note:** the logs will be deleted after 08-Mar-2020

Agent/Stage | Status
-- | --
[_main](https://gist.githubusercontent.com/mellanox-github/a5fbb44189c1fea22032b012a9e5e593/raw/c6556b65eb8fc9050f3a30fd0d0194023a889328/_main) | :heavy_check_mark: SUCCESS
[hpc-arm-cavium-jenkins_W0](https://gist.githubusercontent.com/mellanox-github/a5fbb44189c1fea22032b012a9e5e593/raw/a46f6ed149ad92a7773918304932127107cb3b33/hpc-arm-cavium-jenkins_W0) | :heavy_check_mark: SUCCESS
[hpc-arm-cavium-jenkins_W1](https://gist.githubusercontent.com/mellanox-github/a5fbb44189c1fea22032b012a9e5e593/raw/a873b293a2cc90dc3dc0f23dd46977188118157c/hpc-arm-cavium-jenkins_W1) | :heavy_check_mark: SUCCESS
[hpc-arm-cavium-jenkins_W2](https://gist.githubusercontent.com/mellanox-github/a5fbb44189c1fea22032b012a9e5e593/raw/3a423045c1e1595e675a402aed771a7803a893fd/hpc-arm-cavium-jenkins_W2) | :heavy_check_mark: SUCCESS
[hpc-arm-cavium-jenkins_W3](https://gist.githubusercontent.com/mellanox-github/a5fbb44189c1fea22032b012a9e5e593/raw/9aefbdc532a351aa3651e896dbcb62c0caf9c663/hpc-arm-cavium-jenkins_W3) | :heavy_check_mark: SUCCESS
[hpc-arm-hwi-jenkins_W0](https://gist.githubusercontent.com/mellanox-github/a5fbb44189c1fea22032b012a9e5e593/raw/6954dcb7cb4381e1f800843454ee520372a552e0/hpc-arm-hwi-jenkins_W0) | :heavy_check_mark: SUCCESS
[hpc-arm-hwi-jenkins_W1](https://gist.githubusercontent.com/mellanox-github/a5fbb44189c1fea22032b012a9e5e593/raw/99661126ab2101a2f658da05fe542f8a43e6cfad/hpc-arm-hwi-jenkins_W1) | :heavy_check_mark: SUCCESS
[hpc-arm-hwi-jenkins_W2](https://gist.githubusercontent.com/mellanox-github/a5fbb44189c1fea22032b012a9e5e593/raw/6a40cb109a4c3300744decd660fbc84041b28b49/hpc-arm-hwi-jenkins_W2) | :heavy_check_mark: SUCCESS
[hpc-arm-hwi-jenkins_W3](https://gist.githubusercontent.com/mellanox-github/a5fbb44189c1fea22032b012a9e5e593/raw/6385a60f5b9ee3b3fcf166e7d1c5abeb502e56e9/hpc-arm-hwi-jenkins_W3) | :heavy_check_mark: SUCCESS
[hpc-test-node-gpu_W0](https://gist.githubusercontent.com/mellanox-github/a5fbb44189c1fea22032b012a9e5e593/raw/88498a73293bf7e08c8b01dc218b15b599285878/hpc-test-node-gpu_W0) | :heavy_check_mark: SUCCESS
[hpc-test-node-gpu_W1](https://gist.githubusercontent.com/mellanox-github/a5fbb44189c1fea22032b012a9e5e593/raw/23d98655274c023dcdef4d3bcf742d88a5494711/hpc-test-node-gpu_W1) | :heavy_check_mark: SUCCESS
[hpc-test-node-gpu_W2](https://gist.githubusercontent.com/mellanox-github/a5fbb44189c1fea22032b012a9e5e593/raw/cd17fb043c6f60b218ff5da6a2bce7764635cbac/hpc-test-node-gpu_W2) | :heavy_check_mark: SUCCESS
[hpc-test-node-gpu_W3](https://gist.githubusercontent.com/mellanox-github/a5fbb44189c1fea22032b012a9e5e593/raw/9f8b0ac117aa009f9cc58b2a569bdab4c5b53337/hpc-test-node-gpu_W3) | :heavy_check_mark: SUCCESS
[hpc-test-node-legacy_W0](https://gist.githubusercontent.com/mellanox-github/a5fbb44189c1fea22032b012a9e5e593/raw/14de6a495f35c3ae0ad70d68c62223461ec77b71/hpc-test-node-legacy_W0) | :heavy_check_mark: SUCCESS
[hpc-test-node-legacy_W1](https://gist.githubusercontent.com/mellanox-github/a5fbb44189c1fea22032b012a9e5e593/raw/76f88a33c0835adb6e89ed44395bb41434f1bcd7/hpc-test-node-legacy_W1) | :heavy_check_mark: SUCCESS
[hpc-test-node-legacy_W2](https://gist.githubusercontent.com/mellanox-github/a5fbb44189c1fea22032b012a9e5e593/raw/6358d78c2b8e0060e7ff52e39fd354d433bda36b/hpc-test-node-legacy_W2) | :heavy_check_mark: SUCCESS
[hpc-test-node-legacy_W3](https://gist.githubusercontent.com/mellanox-github/a5fbb44189c1fea22032b012a9e5e593/raw/c17568f709499e0a849e10855370055059a84b9e/hpc-test-node-legacy_W3) | :heavy_check_mark: SUCCESS
[hpc-test-node-new_W0](https://gist.githubusercontent.com/mellanox-github/a5fbb44189c1fea22032b012a9e5e593/raw/557bb92640c442dab632266720966618e3b6b2ba/hpc-test-node-new_W0) | :heavy_check_mark: SUCCESS
[hpc-test-node-new_W1](https://gist.githubusercontent.com/mellanox-github/a5fbb44189c1fea22032b012a9e5e593/raw/c28cfc0188253588b1487abe4aec698ddcdffca7/hpc-test-node-new_W1) | :heavy_check_mark: SUCCESS
[hpc-test-node-new_W2](https://gist.githubusercontent.com/mellanox-github/a5fbb44189c1fea22032b012a9e5e593/raw/dcd9522fceb8d0f31af522fe2429dab9fc50b792/hpc-test-node-new_W2) | :heavy_check_mark: SUCCESS
[hpc-test-node-new_W3](https://gist.githubusercontent.com/mellanox-github/a5fbb44189c1fea22032b012a9e5e593/raw/00de5543069898e75f1c1b95dab9a8d1e340e7c5/hpc-test-node-new_W3) | :heavy_check_mark: SUCCESS
[r-vmb-ppc-jenkins_W0](https://gist.githubusercontent.com/mellanox-github/a5fbb44189c1fea22032b012a9e5e593/raw/243f3b7514aa678973df5668d966f1a45603c57b/r-vmb-ppc-jenkins_W0) | :heavy_check_mark: SUCCESS
[r-vmb-ppc-jenkins_W1](https://gist.githubusercontent.com/mellanox-github/a5fbb44189c1fea22032b012a9e5e593/raw/06991422899d86ccf1e01f4648b3346aa071a654/r-vmb-ppc-jenkins_W1) | :heavy_check_mark: SUCCESS
[r-vmb-ppc-jenkins_W2](https://gist.githubusercontent.com/mellanox-github/a5fbb44189c1fea22032b012a9e5e593/raw/8730134b375bfd7f9d64ee54b8b852bbdc67c1ec/r-vmb-ppc-jenkins_W2) | :heavy_check_mark: SUCCESS
[r-vmb-ppc-jenkins_W3](https://gist.githubusercontent.com/mellanox-github/a5fbb44189c1fea22032b012a9e5e593/raw/356a14fcaad6d3da09b726cbeed550a7881139eb/r-vmb-ppc-jenkins_W3) | :heavy_check_mark: SUCCESS

**Comment by yosefe** [comment]
@Akshay-Venkatesh AFAIK GPU should not be affected, and may even be improved.

IB spec semantics do not define a strict order between data/sge elements, only between CQE and data (which is unaffected by this PR). Still, some applications came to rely on strict order, so it's an "opt-in" flag in IB Verbs.

**Comment by Akshay-Venkatesh** [comment]
Thanks. Maybe something that nccl folks care about. Cc @bureddy

File: src/uct/ib/base/ib_md.c [hunk 6]
```diff
@@ -185,6 +185,11 @@ static ucs_config_field_t uct_ib_md_config_table[] = {
...
+     "Enable relaxed ordering for PCI writes from device to memory, "
```
Comment (drossetti) [inline src/uct/ib/base/ib_md.c]: why only writes? aren't RO reads enabled too ?

Comment (drossetti) [reply src/uct/ib/base/ib_md.c]: in case, I'd just drop "writes from device to memory" and mention "PCIe transactions"

**Pull request merged**